x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/datafeed/DatafeedUpdate.java

```diff
@@ -158,8 +168,16 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeBoolean(true);
             out.writeStringCollection(Collections.emptyList());
         }
-        out.writeOptionalNamedWriteable(query);
-        out.writeOptionalWriteable(aggregations);
+        if (out.getVersion().before(Version.V_7_1_0)) {
+            out.writeNamedWriteable(lazyQueryParser.apply(query, id, new ArrayList<>()));
```

Comment:
The ouput function needs to be `out.writeOptionalNamedWriteable`, as that's what the pre-7.1 code expects.